### DISCUSSION TITLE
Slint performance problem with DumbBufferDisplay  and LinuxFBDisplay

### DISCUSSION THREAD

**Discussion by fluolite** [slint-ui/slint#7030] [General]
In my project, the app window size is 480px*480px.
I run it in different backend.

1.  Linux-x11 + winit opengl renderer, it can run 60fps.
2.  Linuxkms + software renderer (LinuxFBDisplay), it can run 24fps->30fps.
3.  Linuxkms + software renderer (DumbBufferDisplay), it only can run 12fps, and touch event can not responsive some times.

Is that is right？
DumbBufferDisplay performance should not be better than LinuxFBDisplay?
I'm really a little confused.
I hope to get some advice or help. 
Thank you.

**Comment by ogoffart** [comment]
Maybe this is caused by https://github.com/slint-ui/slint/issues/6951 ?

**Reply by fluolite** [reply]
Is that has been found and solved?
Should we see it in the next version  1.9?
It's so exciting!

**Comment by tronical** [comment]
That's very well possible, at least for 3. The fact that the software renderer performs worse than OpenGL rendering is probably expected (difference between 1 and 2), and 12 is half of 24, which is #6951.

**Reply by fluolite** [reply]
I have also noticed this iuuses too.
The fix code has been submitted to the master?
How can I get it and make a testing.

**Reply by tronical** [reply]
I've got a work-in-progress patch at https://github.com/slint-ui/slint/tree/simon/linuxkms-triple-buffer-gbm but unfortunately it doesn't cover DumbBufferDisplay yet.
